test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -1281,6 +1282,168 @@ public void TensorFlowImageClassification()
             }
         }
 
+        [TensorFlowFact]
+        public void TensorFlowImageClassificationEarlyStoppingIncreasing()
+        {
+            string assetsRelativePath = @"assets";
+            string assetsPath = GetAbsolutePath(assetsRelativePath);
+            string imagesDownloadFolderPath = Path.Combine(assetsPath, "inputs",
+                "images");
+
+            //Download the image set and unzip
+            string finalImagesFolderName = DownloadImageSet(
+                imagesDownloadFolderPath);
+
+            string fullImagesetFolderPath = Path.Combine(
+                imagesDownloadFolderPath, finalImagesFolderName);
+
+            MLContext mlContext = new MLContext(seed: 1);
+
+            //Load all the original images info
+            IEnumerable<ImageData> images = LoadImagesFromDirectory(
+                folder: fullImagesetFolderPath, useFolderNameAsLabel: true);
+
+            IDataView shuffledFullImagesDataset = mlContext.Data.ShuffleRows(
+                mlContext.Data.LoadFromEnumerable(images), seed: 1);
+
+            shuffledFullImagesDataset = mlContext.Transforms.Conversion
+                .MapValueToKey("Label")
+                .Fit(shuffledFullImagesDataset)
+                .Transform(shuffledFullImagesDataset);
+
+            // Split the data 80:10 into train and test sets, train and evaluate.
+            TrainTestData trainTestData = mlContext.Data.TrainTestSplit(
+                shuffledFullImagesDataset, testFraction: 0.2, seed: 1);
+
+            IDataView trainDataset = trainTestData.TrainSet;
+            IDataView testDataset = trainTestData.TestSet;
+
+            int lastEpoch = 0;
+            var pipeline = mlContext.Model.ImageClassification(
+                "ImagePath", "Label",
+                arch: ImageClassificationEstimator.Architecture.ResnetV2101,
+                epoch: 50,
+                batchSize: 5,
+                learningRate: 0.01f,
+                earlyStopping: new ImageClassificationEstimator.EarlyStopping(),
```

Comment:
>earlyStopping: new ImageClassificationEstimator.EarlyStopping(), [](start = 16, length = 64)

isn't this the case by default? How is this test different from the above test? May be in the above test disable early stopping so we get that case covered and here enable but also verify the epoch at which it stops via metrics callback

Reply:
done

---
In reply to: [330318498](https://github.com/dotnet/machinelearning/pull/4237#discussion_r330318498) [](ancestors = 330318498)